### PR TITLE
Clarify boundaries of Kubernetes.

### PR DESCRIPTION
**Pull request by bgrant0607**
An attempt to write down what we've discussed a number of times. 

@thockin @brendandburns

### CODE REVIEW

**Comment by thockin** [comment]
lgtm

File: docs/whatisk8s.md [hunk 1]
```diff
@@ -77,6 +79,17 @@ Here are some key points:
...
+* On the other hand, a number of PaaS systems run *on* Kubernetes, such as [Openshift](https://github.com/openshift/origin) and [Deis](http://deis.io/). You could also roll your own custom PaaS, integrate with a CI system of your choice, or get along just fine with just Kubernetes: bring your container images and deploy them on Kubernetes.
+* Since Kubernetes operates at the application level rather than at just the hardware level, it provides some generally applicable features common to PaaS offerings, such as deployment, scaling, load balancing, logging, monitoring, etc. However, Kubernetes is not monolithic, and these default solutions are optional and pluggable.
+
+Kubernetes is not a mere "orchestration system"; it eliminates the need for orchestration:
+* The technical definition of "orchestration" is execution of a defined workflow: do A, then B, then C. In contrast, Kubernetes is compromised of a set of control processes that continuously drive current state towards the provided desired state. It shouldn't matter how you get from A to C: make it so. This results in a system that is easier to use and more powerful, robust, and resilient.
```
Comment (derekwaynecarr) [inline docs/whatisk8s.md]: Kubernetes is "comprised"; it definitely is not compromised ;-)

**Comment by bprashanth** [comment]
kicked shippable for flake (https://github.com/kubernetes/kubernetes/issues/13284)

File: docs/whatisk8s.md (hunk 1 above)
Comment (bgrant0607) [inline docs/whatisk8s.md]: Oops, good catch. Will fix.

**Comment by bgrant0607** [comment]
@k8s-bot test this please

**Pull request merged**